src/main/kotlin/controllers/worldRenderer/Renderer.kt

```diff
@@ -197,154 +194,149 @@ class Renderer constructor(
-        val i =
-            GLBuffers.newDirectIntBuffer(intArrayOf(GL.GL_COLOR_ATTACHMENT0, GL2ES2.GL_COLOR_ATTACHMENT1))
-        gl.glDrawBuffers(2, i)
+        // TODO: was this necessary?
+        val i = BufferUtils.createIntBuffer(2).put(GL_COLOR_ATTACHMENT0).put(GL_COLOR_ATTACHMENT1).flip()
+        glDrawBuffers(i)
 
         // Clear scene
         val sky = 9493480
-        gl.glClearColor((sky shr 16 and 0xFF) / 255f, (sky shr 8 and 0xFF) / 255f, (sky and 0xFF) / 255f, 1f)
-        gl.glClear(GL.GL_COLOR_BUFFER_BIT or GL.GL_DEPTH_BUFFER_BIT)
-
-        modelBuffers.flip()
-        modelBuffers.flipVertUv()
-
-        val vertexBuffer: IntBuffer = modelBuffers.vertexBuffer.buffer
-        val uvBuffer: FloatBuffer = modelBuffers.uvBuffer.buffer
-        val modelBuffer: IntBuffer = modelBuffers.modelBuffer.buffer
-        val modelBufferSmall: IntBuffer = modelBuffers.modelBufferSmall.buffer
-        val modelBufferUnordered: IntBuffer = modelBuffers.modelBufferUnordered.buffer
-        gl.glBindBuffer(GL.GL_ARRAY_BUFFER, tmpBufferId)
-        gl.glBufferData(
-            GL.GL_ARRAY_BUFFER,
-            vertexBuffer.limit() * Integer.BYTES.toLong(),
-            vertexBuffer,
-            GL.GL_DYNAMIC_DRAW
-        )
-        gl.glBindBuffer(GL.GL_ARRAY_BUFFER, tmpUvBufferId)
-        gl.glBufferData(
-            GL.GL_ARRAY_BUFFER,
-            uvBuffer.limit() * java.lang.Float.BYTES.toLong(),
-            uvBuffer,
-            GL.GL_DYNAMIC_DRAW
-        )
-        gl.glBindBuffer(GL.GL_ARRAY_BUFFER, tmpModelBufferId)
-        gl.glBufferData(
-            GL.GL_ARRAY_BUFFER,
-            modelBuffer.limit() * Integer.BYTES.toLong(),
-            modelBuffer,
-            GL.GL_DYNAMIC_DRAW
-        )
-        gl.glBindBuffer(GL.GL_ARRAY_BUFFER, tmpModelBufferSmallId)
-        gl.glBufferData(
-            GL.GL_ARRAY_BUFFER,
-            modelBufferSmall.limit() * Integer.BYTES.toLong(),
-            modelBufferSmall,
-            GL.GL_DYNAMIC_DRAW
-        )
-        gl.glBindBuffer(GL.GL_ARRAY_BUFFER, tmpModelBufferUnorderedId)
-        gl.glBufferData(
-            GL.GL_ARRAY_BUFFER,
-            modelBufferUnordered.limit() * Integer.BYTES.toLong(),
-            modelBufferUnordered,
-            GL.GL_DYNAMIC_DRAW
-        )
+        glClearColor((sky shr 16 and 0xFF) / 255f, (sky shr 8 and 0xFF) / 255f, (sky and 0xFF) / 255f, 1f)
+        glClear(GL_COLOR_BUFFER_BIT or GL_DEPTH_BUFFER_BIT)
+
         val clientCycle = ((System.currentTimeMillis() - clientStart) / 20).toInt() // 50 fps
 
+        priorityRenderer.produceVertices(camera, clientCycle)
+
+        if (textureArrayId == -1) {
+            // lazy init textures as they may not be loaded at plugin start.
+            // this will return -1 and retry if not all textures are loaded yet, too.
+            textureArrayId = textureManager.initTextureArray()
+        }
+//        val textures: Array<TextureDefinition> = textureProvider.getTextureDefinitions()
```

Comment:
Is this caused by the linting?

Reply:
ctrl+slash leaves comment characters at the start (when removing lines) but manual comments tend to be in-line. i'm not too fussed about leaving it this way but if it's a bother I can change it and/or yeet  the code